src/transformers/generation_utils.py

```diff
@@ -904,6 +905,11 @@ def generate(
                 List of token ids that are not allowed to be generated. In order to get the token ids of the words that
                 should not appear in the generated text, use `tokenizer(bad_words, add_prefix_space=True,
                 add_special_tokens=False).input_ids`.
+            force_words_ids(`List[List[int]]` or `List[List[List[int]]]`, *optional*):
+                List of token ids that must be generated. If given a `List[List[int]]`, this is treated as a simple
+                list of words that must be included, the opposite to `bad_words_ids`. If given `List[List[List[int]]]`,
+                this triggers a [disjunctive constraint](https://github.com/huggingface/transformers/issues/14081),
```

Comment:
Good for me for now to link the PR, in a follow-up we could change this to a blog post :-)

Reply:
Yes! Agreed. I'll put in this change once there's a blog post to link to.

Reply:
This behavior depending on the type of the argument seems highly dangerous to me (dangerous in the sense that users won't understand what happens since we're changing the behavior in a non intuitive way).

I am not sure I understand disjunctive constraint, but if it's pick any one of the elements in my list as a `force_ids` then, I would suggest, that we change this to include only in the doc the fact that `List[List[List[int]]]]` And provide an example.

```
force_word_ids = [ 
  [ 
    [ 12, 24] # Miami
    [ 43 , 53] # Toronto
  ] ,
  [ [ 23] # Plane
  ]
]
# Then the result generation will contain exactly one of `Miami` or `Toronto`, and the word `Plane`
```

If we do that it becomes quite clear that having a single option of all `force_word_ids` will force all the words to be used, no ?

If I understand correctly and Disjunctinve is exactly a superset of the other one, then we can even just use it instead of the old `force_word_ids`. Performance might make us want to keep the old code though I am not sure how it works at this point.

Anyway, at this point we should definitely clean the inputs so that we have a consistent code later on. (Like just forcing everything to be `List[List[List[int]]]]`  even if we keep accepting the old format for backward compatibility.